src/EA.Weee.Web/Areas/AatfReturn/Controllers/SelectYourPCSController.cs

```diff
@@ -69,11 +87,140 @@ public virtual async Task<ActionResult> Index(SelectYourPcsViewModel viewModel)
+
+                var existing = await client.SendAsync(User.GetAccessToken(), request);
+
+                SelectYourPcsViewModel viewModel = new SelectYourPcsViewModel
+                {
+                    OrganisationId = organisationId,
+                    ReturnId = returnId,
+                    SchemeList = await client.SendAsync(User.GetAccessToken(), new GetSchemesExternal()),
+                    Reselect = true
+                };
+
+                if (TempData.ContainsKey("selectedSchemes"))
+                {
+                    viewModel.SelectedSchemes = TempData["selectedSchemes"] as List<Guid>;
+                    TempData.Remove("selectedSchemes");
+                }
+                else
+                { 
+                    viewModel.SelectedSchemes = existing.SchemeDataItems.Select(p => p.Id).ToList();
+                }
+               
+                await SetBreadcrumb(organisationId, BreadCrumbConstant.AatfReturn);
+
+                return View("Reselect", viewModel);
+            }
+        }
+
+        [HttpPost]
+        [ValidateAntiForgeryToken]
+        private async Task<ActionResult> Reselect(SelectYourPcsViewModel viewModel)
+        {
+            using (var client = apiClient())
+            {
+                SchemeDataList existing = await client.SendAsync(User.GetAccessToken(), new GetReturnScheme(viewModel.ReturnId));
+
+                if (HaveSchemesBeenRemoved(viewModel, existing.SchemeDataItems.ToList()))
+                {
+                    PcsRemovedViewModel model = new PcsRemovedViewModel()
+                    {
+                        RemovedSchemeList = viewModel.SchemeList.Where(p => viewModel.SelectedSchemes.Contains(p.Id) == false).ToList(),
+                        SelectedSchemes = viewModel.SelectedSchemes,
+                        RemovedSchemes = viewModel.SchemeList.Select(p => p.Id).Where(q => viewModel.SelectedSchemes.Contains(q) == false).ToList(),
+                        ReturnId = viewModel.ReturnId,
+                        OrganisationId = viewModel.OrganisationId
+                    };
+
+                    return View("PcsRemoved", model);
+                }
+
+                return await SaveAndContinue(existing, viewModel.SelectedSchemes, viewModel.ReturnId);
+            }
+        }
+
+        private async Task<ActionResult> SaveAndContinue(SchemeDataList existingSchemes, List<Guid> schemeIdsToAdd, Guid returnId)
+        {
+            using (var client = apiClient())
+            {
+                foreach (SchemeData scheme in existingSchemes.SchemeDataItems)
+                {
+                    if (schemeIdsToAdd.Contains(scheme.Id))
+                    {
+                        schemeIdsToAdd.Remove(scheme.Id);
+                    }
+                }
+
+                foreach (Guid scheme in schemeIdsToAdd)
+                {
+                    AddReturnScheme request = new AddReturnScheme()
```

Comment:
Change this so that only a single request is called... i.e. in case of failure we don't end in an odd state. Request should contain a list of ID's.

Reply:
Changed